cql3/restrictions/multi_column_restriction.hh

```diff
@@ -370,13 +370,13 @@ class multi_column_restriction::slice final : public multi_column_restriction {
         , _mode(m)
     { }
 public:
-    slice(schema_ptr schema, std::vector<const column_definition*> defs, statements::bound bound, bool inclusive, shared_ptr<term> term, mode m = mode::cql)
-        : slice(schema, defs, bounds_slice::new_instance(bound, inclusive, term.get() != nullptr ? std::make_optional(expr::to_expression(term)) : std::nullopt), m)
+    slice(schema_ptr schema, std::vector<const column_definition*> defs, statements::bound bound, bool inclusive, expr::expression e, mode m = mode::cql)
+        : slice(schema, defs, bounds_slice::new_instance(bound, inclusive, e), m)
```

Comment:
Existing code had optional term here, why is it non-optional now?

Reply:
It used to be `optional` because I wasn't sure if the `term` can be `nullptr`.
Now it turns out that there is always a value there, so I removed the `optional`.

The only place where this constructor is called is in `multi_column_relation.hh`:
```c++
auto e = to_expression(col_specs, get_value(), db, schema->ks_name(), ctx);
return ::make_shared<restrictions::multi_column_restriction::slice>(schema, rs, bound, inclusive, std::move(e), _mode);
```
And `to_expression` returns an `expression`

Reply:
ok